frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx

```diff
@@ -74,6 +80,35 @@ export const AddCapacityModal = (props: AddCapacityModalProps) => {
   const usedCapacity = humanizeBinaryBytes(usedCapacityMetric);
   const totalCapacity = humanizeBinaryBytes(totalCapacityMetric);
 
+  const validateSC = React.useCallback(() => {
+    if (!selectedSCName) return t('ceph-storage-plugin~No StorageClass selected');
+    if (!isNoProvionerSC) return '';
+    if (isArbiterEnabled) {
+      return !isArbiterSC(selectedSCName, pvData, nodesData)
+        ? t(
+            'ceph-storage-plugin~The Arbiter stretch cluster requires a minimum of 4 nodes (2 different zones, 2 nodes per zone). Please choose a different storage class or create a new volume set that matches the minimum node requirement.',
+          )
+        : '';
+    }
+    if (!hasFlexibleScaling) {
```

Comment:
This condition aligns with what we want to do -  only for rack based clusters , we want to check further whether added storage class has min 3 rack labels or not.
```suggestion
    if (failureDomain === 'rack'  && !isIntenalAttachedSC) {
```

Reply:
I have few concerns related to this:

1) What if we are adding capacity with `hasFlexibleScaling === true` & `failureDomain === 'rack'` configuration. In this case L93 will be true and it will show wrong error message. So, I will add a check above somewhere that if `hasFlexibleScaling ==true` do not check for `isIntenalAttachedSC` or `isArbiterSC`.

2) "This condition aligns with what we want to do - only for rack based clusters" --> Do we not want to add `isIntenalAttachedSC` validation for AWS provider as well ?? 
Right now, `isIntenalAttachedSC` function handles both situation (Rack and AWS). For AWS provider we are checking if we have 3 nodes with PV on each or not, similar to what we do in 2nd step of wizard flow and if failureDomain is "rack" we check for total racks with PVs attached. So, all situations are handled right now.

Reply:
1. is invalid, flexible scaling sets a host based failure domain
2.Again its only for LSO based deployments on AWS/Baremetal/Vsphere.
Let's call `isIntenalAttachedSC` - validateRacks or something else if thats confusing.

Reply:
1) thanks, yeah u r right, I missed this point.

2) "its only for LSO based deployments on **AWS**/Baremetal/Vsphere" --> that is what I wanted to confirm. For Baremetal/Vsphere it is fine, but for AWS we do not have rack labels, right ?? So, checking `failureDomain === 'rack'` for AWS infrastructure will be always false. So, we do not want to add `isIntenalAttachedSC` validation for AWS case ??

Reply:
For AWS as well we expect to check. 
The nodes should be in three different racks/topologies - its for all LSO based deployment.